src/types/error.js

```diff
@@ -264,19 +264,45 @@ export class SwapCurrencyError extends Error {
   +pluginId: string
   +fromCurrency: string
   +toCurrency: string
+  +fromTokenId: string | void
+  +toTokenId: string | void
 
   constructor(
     swapInfo: EdgeSwapInfo,
-    fromCurrency: string,
-    toCurrency: string
+    // Passing currency codes is deprecated:
+    request: string | EdgeSwapRequest,
+    toCurrency?: string
   ) {
+    // Backwards-compatible currency code extraction:
+    let fromCurrency: string
+    if (typeof request === 'string') {
+      fromCurrency = request
+      if (toCurrency == null) toCurrency = 'unknown'
+    } else {
+      fromCurrency =
+        request.fromTokenId != null
+          ? request.fromWallet.currencyConfig.allTokens[request.fromTokenId]
+              .currencyCode
+          : request.fromCurrencyCode ??
+            request.fromWallet.currencyInfo.currencyCode
```

Comment:
why fallback to `request.fromWallet.currencyInfo.currencyCode`? Shouldn't the request.fromCurrencyCode always exist?

Reply:
The type system demands it. But I could put `''` or something if that would be simpler.

Reply:
I think some invalid value would be better so we don't imply that is an expected code-flow. `` would be fine.